### PR TITLE
add non-interactive mode to open-group-channel

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Added non-interactive mode to open-group-channel

https://github.com/alexbosworth/paid-services/pull/9

Files changed:
bos

### CODE REVIEW

File: bos [hunk 1]
```diff
@@ -1201,14 +1201,24 @@ prog
   .command('open-group-channel', 'Open balanced channels with multiple peers')
   .help('Remote nodes must be prepared to participate in this type of channel')
   .help('The remote node should also run this same command to join the group')
+  .option('--capacity <channel_capacity>', 'Total capacity of the channels', INT)
+  .option('--fee-rate <per_vbyte>', 'Fee rate for open (Max 10 sats/vByte)', INT)
```
Comment (alexbosworth) [inline bos:1205]:
```suggestion
  .option('--fee-rate <rate>', 'Fee rate for open (Default 10/vByte)', INT)
```

File: bos [hunk 2]
```diff
@@ -1201,14 +1201,24 @@ prog
...
+  .option('--invite-code <group_invite_code>', 'Invite code to join a group', STRING)
   .option('--node <node_name>', 'Use saved node details instead of local node')
+  .option('--size <group_size', 'Group size', INT)
```
Comment (alexbosworth) [inline bos:1208]:
```suggestion
  .option('--size <group_size', 'Create new group with number of members', INT)
```

File: bos [hunk 3]
```diff
@@ -1143,7 +1143,7 @@ prog
   .option('--node <node_name>', 'Saved node to open channels')
   .option('--opening-node <node_name>', 'Open with saved node', REPEATABLE)
   .option('--set-fee-rate <ppm>', 'Set forward fee rate to peer', REPEATABLE)
-  .option('--type <type>', 'Type of channel (private/public)', REPEATABLE)
+  .option('--type <type>', 'Type of channel (private/private-trusted/public/public-trusted)', REPEATABLE)
```
Comment (alexbosworth) [inline bos:1146]:
```suggestion
  .option('--type <type>', 'Type of channel (private/public)', REPEATABLE)
```

**Comment by niteshbalusu11** [comment]
Added a commit to add new commands on bos file

File: bos [hunk 4]
```diff
@@ -518,6 +518,32 @@ prog
...
 
+    // Open a group channel with multiple peers
+    .command('create-group-channel', 'Create group to open balanced channels with multiple peers')
+    .visible(false)
+    .help('Remote nodes must be prepared to participate in this type of channel')
+    .help('The remote node should run join-group-channel to join the group')
+    .option('--capacity <channel_capacity>', 'Total capacity of the channels', INT, 5e6)
+    .option('--fee-rate <per_vbyte>', 'Fee rate for open (Default 5/vByte)', INT, 5)
+    .option('--node <node_name>', 'Use saved node details instead of local node')
+    .option('--size <group_size', 'Create new group with number of members', INT)
```
Comment (alexbosworth) [inline bos:529]:
```suggestion
    .option('--size <group_size>', 'Create new group with number of members', INT)
```

File: bos [hunk 5]
```diff
@@ -1018,6 +1044,30 @@ prog
     });
   })
 
+  // Open a group channel with multiple peers
+  .command('join-group-channel', 'Join a group to open balanced channels with multiple peers')
+  .visible(false)
+  .help('Remote nodes must be prepared to participate in this type of channel')
+  .help('The initiator node should run create-group-channel to create the group')
+  .option('--invite-code <invite_code>', 'Invite code to join a group', STRING)
+  .option('--max-fee-rate <per_vbyte>', 'Maximum allowed rate for open (Default 5 sats/vByte)', INT, 5)
```
Comment (alexbosworth) [inline bos:1053]:
```suggestion
  .option('--max-fee-rate <per_vbyte>', 'Maximum allowed rate for open (Default 5/vByte)', INT, 5)
```

File: bos [hunk 6]
```diff
@@ -518,6 +518,32 @@ prog
     });
   })
 
+    // Open a group channel with multiple peers
+    .command('create-group-channel', 'Create group to open balanced channels with multiple peers')
+    .visible(false)
+    .help('Remote nodes must be prepared to participate in this type of channel')
+    .help('The remote node should run join-group-channel to join the group')
+    .option('--capacity <channel_capacity>', 'Total capacity of the channels', INT, 5e6)
```
Comment (alexbosworth) [inline bos:526]:
```suggestion
    .option('--capacity <channel_capacity>', 'Channel capacity', INT, 5e6)
```

File: bos [hunk 7]
```diff
@@ -518,6 +518,32 @@ prog
...
+    .option('--fee-rate <per_vbyte>', 'Fee rate for open (Default 5/vByte)', INT, 5)
```
Comment (alexbosworth) [inline bos:527]:
```suggestion
    .option('--fee-rate <per_vbyte>', 'Chain fee rate for open', INT, 5)
```

I think the help already shows the default

File: bos [hunk 8]
```diff
@@ -518,6 +518,32 @@ prog
...
+    .option('--node <node_name>', 'Use saved node details instead of local node')
+    .option('--size <group_size>', 'Create new group with number of members', INT)
+    .action((args, options, logger) => {
+      return new Promise(async (resolve, reject) => {
+        try {
+          return paidServices.createGroupChannel({
```
Comment (alexbosworth) [inline bos:533]:
```suggestion
          return await paidServices.createGroupChannel({
```

File: bos [hunk 9]
```diff
@@ -1018,6 +1044,30 @@ prog
...
+  .visible(false)
+  .help('Remote nodes must be prepared to participate in this type of channel')
+  .help('The initiator node should run create-group-channel to create the group')
+  .option('--invite-code <invite_code>', 'Invite code to join a group', STRING)
+  .option('--max-fee-rate <per_vbyte>', 'Maximum allowed rate for open (Default 5/vByte)', INT, 5)
+  .option('--node <node_name>', 'Use saved node details instead of local node')
+  .action((args, options, logger) => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        return paidServices.joinGroupChannel({
```
Comment (alexbosworth) [inline bos:1058]:
```suggestion
        return await paidServices.joinGroupChannel({
```

File: bos [hunk 10]
```diff
@@ -1018,6 +1044,30 @@ prog
     });
   })
 
+  // Open a group channel with multiple peers
+  .command('join-group-channel', 'Join a group to open balanced channels with multiple peers')
+  .visible(false)
+  .help('Remote nodes must be prepared to participate in this type of channel')
+  .help('The initiator node should run create-group-channel to create the group')
+  .option('--invite-code <invite_code>', 'Invite code to join a group', STRING)
+  .option('--max-fee-rate <per_vbyte>', 'Maximum allowed rate for open (Default 5/vByte)', INT, 5)
```
Comment (alexbosworth) [inline bos:1053]:
```suggestion
  .option('--max-fee-rate <per_vbyte>', 'Maximum chain rate/vbyte for open', INT, 5)
```

**Pull request closed without merging**